django/views/generic/dates.py

```diff
@@ -245,8 +247,17 @@ def uses_datetime_field(self):
         if it's a `DateField`.
         """
         model = self.get_queryset().model if self.model is None else self.model
-        field = model._meta.get_field(self.get_date_field())
-        return isinstance(field, models.DateTimeField)
+
+        pieces = self.get_date_field().split(LOOKUP_SEP)
+        fields = []
+        for piece in pieces:
+            parent = model
+            if (fields and fields[-1].is_relation and
```

Comment:
This code isn't very understandable. It seems a helper function, method, or something is needed.